addons/web_editor/controllers/main.py

```diff
@@ -540,14 +541,27 @@ def _update_svg_colors(self, options, svg):
             '4': '#FFFFFF',
             '5': '#383E45',
         }
-
+        bundle_css = None
+        regex_hex = r'#[0-9A-F]{6,8}'
+        regex_rgba = r'rgba?\(\d{1,3},\d{1,3},\d{1,3}(?:,[0-9.]{1,4})?\)'
         for key, value in options.items():
             colorMatch = re.match('^c([1-5])$', key)
             if colorMatch:
+                css_color_value = value
                 # Check that color is hex or rgb(a) to prevent arbitrary injection
-                if not re.match(r'(?i)^#[0-9A-F]{6,8}$|^rgba?\(\d{1,3},\d{1,3},\d{1,3}(?:,[0-9.]{1,4})?\)$', value.replace(' ', '')):
-                    raise werkzeug.exceptions.BadRequest()
-                user_colors.append([tools.html_escape(value), colorMatch.group(1)])
+                if not re.match(r'(?i)^%s$|^%s$' % (regex_hex, regex_rgba), css_color_value.replace(' ', '')):
+                    if re.match('^o-color-([1-5])$', css_color_value):
+                        if not bundle_css:
+                            bundle = 'web.assets_frontend'
+                            files, _ = request.env["ir.qweb"]._get_asset_content(bundle, options=request.context)
+                            asset = AssetsBundle(bundle, files)
+                            bundle_css = asset.css().index_content
+                        color_search = re.search(r'(?i)--%s:\s+(%s|%s)' % (css_color_value, regex_hex, regex_rgba), bundle_css)
+                        if color_search:
+                            css_color_value = color_search.group(1)
+                    else:
+                        raise werkzeug.exceptions.BadRequest()
+                user_colors.append([tools.html_escape(css_color_value), colorMatch.group(1)])
```

Comment:
Seems good :+1: but maybe a small problem: when you search for a color in the bundle and that you don't find it (`if color_search` is falsy), you should return a bad request too I think?